src/Akka.Persistence.Sql/Snapshot/ByteArraySnapshotDao.cs

```diff
@@ -5,20 +5,77 @@
 // -----------------------------------------------------------------------
 
 using System;
+using System.Collections.Generic;
 using System.Data;
 using System.Linq;
 using System.Threading;
+using System.Threading.Channels;
 using System.Threading.Tasks;
 using Akka.Event;
 using Akka.Persistence.Sql.Config;
 using Akka.Persistence.Sql.Db;
 using Akka.Persistence.Sql.Extensions;
 using Akka.Streams;
+using Akka.Streams.Dsl;
 using Akka.Util;
 using LinqToDB;
+using LinqToDB.Tools;
 
 namespace Akka.Persistence.Sql.Snapshot
 {
+    public static class SubFlowExtensions
+    {
+        public static Source<TOut, TMat> MergeSubStreamsAsSource<TOut, TMat,
+            TClosed>(this SubFlow<TOut, TMat, TClosed> subFlow)
+        {
+            return (Source<TOut,TMat>)(subFlow.MergeSubstreams());
+        }
+    }
```

Comment:
Happy for feedback on this method but TBH it would be nice if some form (even if it's not returning `Source<TOut,TMat>`) was put into akka streams proper. It's _really_ useful